File: LC-1281/LC-1281.js
/**
 * @param {number} n
 * @return {number}
 */
var subtractProductAndSum = function (n) {
  /**
   * Naive Approachs
   */
  let num = String(n),
    product = 1,
    digitSum = 0

  for (let i = 0; i < num.length; ++i) {
    product *= num[i] - '0'
    digitSum += num[i] - '0'
  }

  return product - digitSum
}

File: LC-1281/LC-1281.cpp
class Solution {
public:
    /**
     * Concepts: Naive Approach
     */
    int subtractProductAndSum(int n) {
        int product = 1, digitSum = 0;
        while (n != 0) {
            product *= (n % 10);
            digitSum += (n % 10);
            n /= 10;
        }
        return product - digitSum;
    }
};